src/decoder/stream.rs

```diff
@@ -1227,35 +1136,23 @@ impl StreamingDecoder {
             ColorType::Indexed => {
                 // The transparency chunk must be after the palette chunk and
                 // before the data chunk.
-                if info.palette.is_none() {
-                    return Err(DecodingError::Format(
-                        FormatErrorInner::BeforePlte { kind: chunk::tRNS }.into(),
-                    ));
-                } else if self.have_idat {
-                    return Err(DecodingError::Format(
-                        FormatErrorInner::OutsidePlteIdat { kind: chunk::tRNS }.into(),
-                    ));
+                if info.palette.is_some() || !self.have_idat {
```

Comment:
Preserving the old conditions requires saying `&&` instead of `||`:  `info.palette.is_some() && !self.have_idat` .  I'll fix this when rebasing... :-/